components/locale/src/subtags/language.rs

```diff
@@ -67,6 +67,53 @@ impl Language {
         }
     }
 
+    /// Deconstructs the `Language` into raw format to be consumed
+    /// by `from_raw_unchecked`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_locale::subtags::Language;
+    ///
+    /// let lang = Language::from_bytes(b"en")
+    ///     .expect("Parsing failed.");
+    ///
+    /// let raw = lang.into_raw();
+    /// let lang = unsafe { Language::from_raw_unchecked(raw) };
+    /// assert_eq!(lang, "en");
+    /// ```
+    pub fn into_raw(self) -> Option<u64> {
```

Comment:
Here and elsewhere: It seems odd to return a `u64`.  Returning the actual `TinyStr8` would have more type safety, and then you can call another method to convert the `TinyStr8` into the raw untyped value if you so desire.

If you make this change, then `from_raw_unchecked` should also be changed to take a `TinyStr8`.

Consider naming the functions `into_tinystr()` and `from_tinystr_unchecked()`.

I think this would be better because it emphasizes that there are two lossy/unsafe operations:

1. Language subtag <-> TinyStr8
2. TinyStr8 <-> u64

All language subtags are TinyStr8s, and all TinyStr8s are u64s, but the reverse is not true in either case.

Reply:
@Manishearth - I remember when we worked on the macros originally for `unic-locale`, you recommended rqw `u64/u32` for raw. Do you have an opinion on switching it to `TinyStr` for reasons laid out by @sffc?

My primary consideration is whether its ok to *expose* the `TinyStr` (which so far has been an internal detail) and stabilize around it even in the "non-meant-for-public-consumption" case of from/to_raw.

WDYT?

Reply:
I think if we are exposing the raw data in the public API, we are exposing the TinyStr _format_ even if not the TinyStr _type_, so we're halfway there, anyway. I think it makes more sense to return/accept TinyStr here.